salt/states/file.py

```diff
@@ -2670,6 +2690,72 @@ def managed(name,
         return _error(
             ret, 'Defaults must be formed as a dict')
 
+    if selinux:
+        current_seuser, current_serole, current_setype, current_serange = __salt__[
+            'file.get_selinux_context'](name).split(':')
+        log.debug('Current selinux context user:{0} role:{1} type:{2} range:{3}'.format(current_seuser, current_serole, current_setype, current_serange))
+        requested_seuser = None
+        requested_serole = None
+        requested_setype = None
+        requested_serange = None
+        # Only set new selinux variables if updates are needed
+        if selinux.get('seuser', None) and selinux['seuser'] != current_seuser:
+            requested_seuser = selinux['seuser']
+        if selinux.get('serole', None) and selinux['serole'] != current_serole:
+            requested_serole = selinux['serole']
+        if selinux.get('setype', None) and selinux['setype'] != current_setype:
+            requested_setype = selinux['setype']
+        if selinux.get('serange', None) and selinux['serange'] != current_serange:
+            requested_serange = selinux['serange']
+
+        if requested_seuser or requested_serole or requested_setype or requested_serange:
+            # selinux updates needed, prep changes output
+            selinux_change_new = ''
+            selinux_change_orig = ''
+            if requested_seuser:
+                selinux_change_new += "User: {0} ".format(requested_seuser)
+                selinux_change_orig += "User: {0} ".format(current_seuser)
+            if requested_serole:
+                selinux_change_new += "Role: {0} ".format(requested_serole)
+                selinux_change_orig += "Role: {0} ".format(current_serole)
+            if requested_setype:
+                selinux_change_new += "Type: {0} ".format(requested_setype)
+                selinux_change_orig += "Type: {0} ".format(current_setype)
+            if requested_serange:
+                selinux_change_new += "Range: {0} ".format(requested_serange)
+                selinux_change_orig += "Range: {0} ".format(current_serange)
+
+            if __opts__['test']:
+                ret['comment'] = 'File {0} selinux context to be updated'.format(name)
+                ret['result'] = None
+                ret['changes'] = {'selinux context': {'Old': selinux_change_orig,
```

Comment:
Here are the reported changes with `test=True`

Reply:
thanks!